pkg/apicheck/check.go

```diff
@@ -24,13 +24,16 @@ import (
 	"github.com/medik8s/self-node-remediation/pkg/reboot"
 )
 
+const maxTimeForNoPeersResponse = 60 * time.Second
```

Comment:
Can you elaborate why one minute?  
Please add explanation as a comment next to it for future reference whether it's a rule of thumb or it has more meaningful explanation.

Reply:
Well in the test I've done (when recreating the bug) I saw the the max time I'm reaching is about 20 seconds.
60 sec seems on the one hand as a reasonable buffer and on the other hand not too long a duration to delay remediation use case of isolated node when it's actually needed

Reply:
- what about adding this to ApiConnectivityCheckConfig?
- doesn't this need to go into the minTimeToAssumeNodeRebooted calculation?

Reply:
>  the max time I'm reaching is about 20 seconds

What exactly is happening or not happening during these 20 seconds?
I'm asking because we already wait a minute before asking peers + rebooting (`MaxErrorsThreshold` * (`CheckInterval` + `ApiServerTimeout`). This change adds another minute, while everyone is asking us to be quicker.

Reply:
>what about adding this to ApiConnectivityCheckConfig?

Agreed :+1: , however I still prefer not to expose this to the user.

>doesn't this need to go into the minTimeToAssumeNodeRebooted calculation?

This is a very good question which indeed I haven't initially considered.
If we want to stay on the safe side we probably should, however this use-case is relevant only for a worker node which is completely isolated (which I assume is pretty rare) so maybe we can discuss other alternatives. 

>What exactly is happening or not happening during these 20 seconds?

For now the only way I've managed to recreate this bug is by taking down 2 out of 3 of the control-plane nodes, once I take 2 control-plane nodes out I witness a short time where worker nodes can't communicate to each other (TBH I don't really know why).

Reply:
Is this somehow aligned with the `--node-monitor-grace-period` duration option for the kube-controller-manager?
If those 2 timeouts are misaligned can we have rebooting worker nodes that are still not considered unhealthy from K8s perspective.
I don't know how often cluster admins tend to change this option and in what direction is the change.

Reply:
> which I assume is pretty rare

even when it's rare, IMHO we shouldn't risk that fencing didn't finish yet when we start deleting pods, it can break workloads

Reply:
>Is this somehow aligned with the --node-monitor-grace-period duration option for the kube-controller-manager?

I don't think so,  IIUC `node-monitor-grace-period` would affect the time it takes for the node to be marked as unhealthy ,where this discussion is about what happens after that.

Reply:
>even when it's rare, IMHO we shouldn't risk that fencing didn't finish yet when we start deleting pods, it can break workloads

Agreed.
How about using a 30 second timeout (instead of 1 minute) and adding it to the min time calculation ?
This way maybe we has less certainty of always resolving the bug but at least we are still making it less likely to happen and we are not hurting the overall time as much. 

I can think of other alternatives of calculating the minTime during runtime according to the type of remediation (i.e different calculation for the use-case of isolated node) but it seems rather complex and IMO would be an overkill for this.